tests/snuba/api/endpoints/test_organization_events_stats.py

```diff
@@ -19,7 +20,45 @@
 from sentry.utils.samples import load_data
 
 
-class OrganizationEventsStatsEndpointTest(APITestCase, SnubaTestCase):
+class OrgStatsBaseTest:
+    def reverse_url(self, org_slug=None):
+        return reverse(
+            self.endpoint,
+            kwargs={"organization_slug": org_slug},
+        )
+
+    def do_request(self, data, features=None, org_slug=None, **kwargs):
+        if features is None:
+            features = {"organizations:discover-basic": True}
+        features.update(self.features)
+        with self.feature(features):
+            if org_slug is None:
+                org_slug = self.organization.slug
+            url = self.reverse_url(org_slug=org_slug)
+            return self.client.get(url, data=data, format="json", **kwargs)
+
+
+class CustomerOrgStatsBaseTest:
+    endpoint = "sentry-api-0-region-organization-events-stats"
+
+    def reverse_url(self, org_slug=None):
+        return reverse(self.endpoint)
+
+    def do_request(
+        self,
+        data,
+        features=None,
+        org_slug=None,
+    ):
+        if org_slug is None:
+            org_slug = self.organization.slug
+        HTTP_HOST = generate_organization_hostname(org_slug)
+        return super().do_request(
+            data=data, features=features, org_slug=org_slug, HTTP_HOST=HTTP_HOST
+        )
```

Comment:
Should we be add a method to our base classes for this? Having each class have a custom do_request() method isn't ideal. Perhaps we could have an interface similar to `do_request()`? But with a name like `do_org_request()` :shrug: